localstack-core/localstack/services/lambda_/invocation/logs.py

```diff
@@ -50,10 +50,34 @@ def run_log_loop(self, *args, **kwargs) -> None:
             log_item = self.log_queue.get()
             if log_item is QUEUE_SHUTDOWN:
                 return
+            # we need to split by newline - but keep the newlines in the strings
+            # strips empty lines, as they are not accepted by cloudwatch
+            logs = [line + "\n" for line in log_item.logs.split("\n") if line]
```

Comment:
Following [this discussion](https://github.com/localstack/localstack/pull/12470/files#diff-2a8c6622efa5103d5807b4619b06e1f9492dc879f3474eee5fdfdf1e96f10a16R505), do we have an intuition where these replacements came from?

```
log_output = log_output.replace("\\x1b", "\n\\x1b")
log_output = log_output.replace("\x1b", "\n\x1b")
```

I just wanted to add the reference in case we need to add them reactively because we removed them without knowing where they came from 😄

Reply:
No, still no idea. Since those are ANSI escape sequences, I still think they stem from the time were we captured the lambda logs directly from docker running inside a full terminal instance.